File: dotnet/ReportGenerator-VB/Machine.vb
Namespace ReportGeneratorVB
    Public Class Machine
        Private _name As String
        Private _bin As String
        Private _location As String

        Public Sub New(ByVal name As String, ByVal location As String)
            Me._name = name
            Me._location = location
        End Sub

        Public Function Take() As String
            Dim result As String = _bin
            _bin = Nothing
            Return result
        End Function

        Public Function Bin() As String
            Return _bin
        End Function

        Public Sub Put(ByVal bin As String)
            Me._bin = bin
        End Sub

        Public Function Name() As String
            Return _name
        End Function
    End Class
End Namespace

File: production-line/csharp/ProductionLine/Machine.cs
namespace ProductionLine
{
    public class Machine
    {
        private string name;
        private Bin bin;
        private string location;

        public Machine(string name, string location)
        {
            this.name = name;
            this.location = location;
        }

        public Bin Take()
        {
            var result = bin;
            bin = null;
            return result;
        }

        public Bin Bin()
        {
            return bin;
        }

        public string Location()
        {
            return location;
        }

        public void Put(Bin bin)
        {
            this.bin = bin;
        }

        public string Name()
        {
            return name;
        }
    }
}
